Core/Services/StitchCoreFCMService/Sources/StitchCoreFCMService/FCMSendMessageNotification.swift

```diff
@@ -0,0 +1,229 @@
+import Foundation
+
+/**
+ * An FCMSendMessageNotification encapsulates the details of an FCM send message request notification payload.
+ */
+public struct FCMSendMessageNotification: Encodable {
+    /**
+     * The notification's title.
+     */
+    public let title: String?
+    
+    /**
+     * The notification's body text.
+     */
+    public let body: String?
+    
+    /**
+     * The sound to play when the device receives the notification.
+     */
+    public let sound: String?
+    
+    /**
+     * The action associated with a user click on the notification.
+     */
+    public let clickAction: String?
+    
+    /**
+     * The the key to the body string in the app's string resources to use to localize the body
+     * text to the user's current localization.
+     */
+    public let bodyLocKey: String?
+    
+    /**
+     * The variable string values to be used in place of the format specifiers in
+     * bodyLocKey to use to localize the body text to the user's current localization.
+     */
+    public let bodyLocArgs: String?
+    
+    /**
+     * The key to the title string in the app's string resources to use to localize the
+     * title text to the user's current localization.
+     */
+    public let titleLocKey: String?
+    
+    /**
+     * The variable string values to be used in place of the format specifiers in
+     * titleLocKey to use to localize the title text to the user's current localization.
+     */
+    public let titleLocArgs: String?
+    
+    /**
+     * The notification's icon. Note: for messages to Android devices only.
+     */
+    public let icon: String?
+    
+    /**
+     * The identifier used to replace existing notifications in the notification drawer.
+     * Note: for messages to Android devices only.
+     */
+    public let tag: String?
+    
+    /**
+     * The notification's icon color, expressed in #rrggbb format. Note: for messages to
+     * Android devices only.
+     */
+    public let color: String?
+    
+    /**
+     * The value of the badge on the home screen app icon. Note: for messages to iOS devices only.
+     */
+    public let badge: String?
+}
+
+
+public class FCMSendMessageNotificationBuilder {
```

Comment:
[nit][maybe] To be honest, all of these builders should probably be structs, so that we can take advantage of the `mutating` keyword. Of course, then we can't inherit from them, but it would be semantically more accurate. I'll leave this one up to you.

Reply:
agreed that it would be semantically nice, but as we discussed, this would break from the semantics of our other builders, so I'll leave as is.